js/battle.js

```diff
@@ -1802,6 +1804,41 @@ var Side = (function () {
 		pokemon.statbarElem = this.battle.statElem.children().last();
 		this.updateStatbar(pokemon, true);
 		pokemon.side.updateSidebar();
+
+		if (pokemon.replacedIllusion && !this.battle.hackedPokemon) {
+			if (this.pokemon.indexOf(pokemon) < 0) {
+				// if the pokemon that is switching in isn't in the team, match it to one that is
+				for (var i = 0; i < this.pokemon.length; i++) {
+					if (pokemon.searchid === this.pokemon[i].searchid) {
+						pokemon.fainted = this.pokemon[i].fainted;
+						this.pokemon[i] = pokemon;
+						break;
+					}
+				}
+			}
+			if (pokemon.fainted) {
```

Comment:
This will also never be true, because this is after `getPokemon` overwrites the old faint status.

Reply:
It's related to the above regarding the `newPokemon` call in `getPokemon`, but right now it doesn't override the faint status (though, that might what is intended?)